deploy/charts/rook-ceph/templates/resources.yaml

```diff
@@ -2213,6 +2221,11 @@ spec:
                       type: object
                   type: object
                   x-kubernetes-preserve-unknown-fields: true
+                  x-kubernetes-validations:
```

Comment:
As a way to test these validations, it would be really nice to have a folder of examples that fail these checks. Then we can have an integration test that verifies they all fail. This could be a follow-up PR.

Reply:
will open issue to track this.

Reply:
https://github.com/rook/rook/issues/13229#issue-1999171755